### PR TITLE
chore: upgrade graphql and enable typing for GT.Field

### PR DESCRIPTION
**Pull request by samerbuna**

### CODE REVIEW

File: src/graphql/admin/root/mutation/account-update-level.ts
Comment (nicolasburtey) [inline src/graphql/admin/root/mutation/account-update-level.ts:23]: the third object is the type of input?

Comment (samerbuna) [reply src/graphql/admin/root/mutation/account-update-level.ts:23]: Yep. It's the type of the `args` object.

Comment (samerbuna) [reply src/graphql/admin/root/mutation/account-update-level.ts]: Ok, I made args the first argument since the others can be set default.

So the order now is TArgs, TSource, TContext with the TSource defaulting to null and TContext defaulting to GraphQLContext

We have a ton of errors already. Will work through them soon.

**Comment by nicolasburtey** [comment]
I have done a rebase @samerbuna

File: src/graphql/types/object/wallet-contact.ts
Comment (nicolasburtey) [inline src/graphql/types/object/wallet-contact.ts:12]: shouldn't `GraphQLContextForUser` be the third argument?

Comment (samerbuna) [reply src/graphql/types/object/wallet-contact.ts:12]: Nope. You don't define the args for resolvers here. We can use `GT.Field` to define args for a field, which we're doing for the mutations, but not queries (yet)

File: src/app/index.types.d.ts
Comment (nicolasburtey) [inline src/app/index.types.d.ts:7]: what is the goal of this type?

File: src/graphql/root/mutation/ln-noamount-invoice-payment-send.ts
Comment (dolcalmi) [inline src/graphql/root/mutation/ln-noamount-invoice-payment-send.ts:50]: why do we need to remove the  for? we are not doing additional logic

**Review by dolcalmi — APPROVED** [review]
Just a small comment but not a blocker. Also I think this need another approval before merge. We can trust in graphql tests but we will need to do additional testing with the app in staging

File: src/graphql/root/mutation/ln-noamount-invoice-payment-send.ts
Comment (nicolasburtey) [reply src/graphql/root/mutation/ln-noamount-invoice-payment-send.ts:50]: typescript doesn't seem smart enough to handle the type from the loop?

Comment (nicolasburtey) [reply src/graphql/root/mutation/ln-noamount-invoice-payment-send.ts:50]: image

**Review by nicolasburtey — APPROVED** [review]

File: src/graphql/root/mutation/ln-noamount-invoice-payment-send.ts
Comment (samerbuna) [reply src/graphql/root/mutation/ln-noamount-invoice-payment-send.ts:50]: Yep

File: src/app/index.types.d.ts
Comment (samerbuna) [reply src/app/index.types.d.ts:7]: I used this in the language scalar.

**Pull request merged**
